### PR TITLE
Add Comment/ Approve /Deny actions

### PR DESCRIPTION
**Pull request by lgalis**
Add submit implementation for the Comment/ Approve /Deny modals, redux actions.
The buttons for these actions are now only available for requests that are still active.

Screenshot from 2019-04-25 06-51-40

Also on the detail page:
1. Active Request

Screenshot from 2019-04-25 13-12-24

2. Finished Request:
Screenshot from 2019-04-25 13-12-37

### CODE REVIEW

**Comment by gmcculloug** [comment]
cc @bzwei

**Comment by lgalis** [comment]
@martinpovolny  - would you be able to review this PR?

Thank you!

File: src/smart-components/request/request-detail/stage.js [hunk 1]
```diff
@@ -78,29 +79,31 @@ class Stage extends Component {
...
           <span style={ { textTransform: 'capitalize' } } id={ `${item.id}-state` }>{ `${item.state}` } </span>
         </DataListCell>
-        <DataListCell>
+        { requestActive && <DataListCell>
           <Level>
             <LevelItem>
               <Link to={ `/requests/detail/${item.request_id}/approve` }>
                 <Button variant="link" aria-label="Approve Request">
-                  Approve
+                    Approve
```
Comment (martinpovolny) [inline src/smart-components/request/request-detail/stage.js]: why are you changing the indentation from 2 to 4 here (and also below)?

File: src/smart-components/request/request-table-helpers.js [hunk 2]
```diff
@@ -5,17 +5,19 @@ import { Button } from '@patternfly/react-core';
...
-      timeAgo(created_at), timeAgo(updated_at), state, decision ]
+        <Button variant="link"> { id } </Button></Link></Fragment>, requester,
+      timeAgo(created_at), timeAgo(updated_at), `${active_stage} of ${total_stages}`, decision ]
     }, {
       parent: key * 2,
-      cells: [{ title: <ExpandableContent id={ id } content={ content } /> }]
+      cells: [{ title: <ExpandableContent id={ id } content={ content } state={ state } active_stage={ active_stage }/> }]
     }
   ]), []);
+};
```
Comment (martinpovolny) [inline src/smart-components/request/request-table-helpers.js]: why is this bracket `}` and the bracket `{` at line 8 added + the `return` statement added on line 9?

File: src/smart-components/request/requests.js [hunk 3]
```diff
@@ -27,19 +28,31 @@ const Requests = ({ fetchRequests, requests, pagination, history }) => {
...
-  const actionResolver = (requestData, { rowIndex }) => rowIndex === 1 ? null :
-    [
-      {
-        title: 'Comment',
-        onClick: () =>
-          history.push(`/requests/add_comment/${requestData.id}`)
-      }
-    ];
+  const actionResolver = (requestData, { rowIndex }) => {
+    return (rowIndex === 1 || areActionsDisabled(requestData) ? null :
```
Comment (martinpovolny) [inline src/smart-components/request/requests.js]: this looks like an extra `return` statement too

File: src/smart-components/request/requests.js [hunk 4]
```diff
@@ -27,19 +28,31 @@ const Requests = ({ fetchRequests, requests, pagination, history }) => {
...
+        {
+          title: 'Comment',
+          onClick: () =>
+            history.push(`/requests/add_comment/${requestData.id}`)
+        }
+      ]);
+
+  };
+
+  const areActionsDisabled = (requestData) => { return !isRequestStateActive(requestData.state);};
```
Comment (martinpovolny) [inline src/smart-components/request/requests.js]: this should be moved higher, abov `actionResolver` (even the linter is complaining about that)

**Comment by martinpovolny** [comment]
The visuals are on par with the UX? I am asking because in the 1st picture, all the data in a row alligned to the left, the white space to the right looks strange.

**Comment by lgalis** [comment]
@martinpovolny  - the visuals are aligned to the UX  mocks - the data I have is is short for each line in the first screenshot.

File: src/smart-components/request/requests.js (hunk 3 above)
Comment (lgalis) [reply src/smart-components/request/requests.js]: I think the return is required here.

File: src/smart-components/request/requests.js (hunk 4 above)
Comment (lgalis) [reply src/smart-components/request/requests.js]: moved up

File: src/smart-components/request/request-table-helpers.js (hunk 2 above)
Comment (lgalis) [reply src/smart-components/request/request-table-helpers.js]: I had other statements here so the curly braces and the return were necessary - but those statements were removed so the brackets/return are no longer necessary - I will remove them.

File: src/smart-components/request/request-detail/stage.js (hunk 1 above)
Comment (lgalis) [reply src/smart-components/request/request-detail/stage.js]: updated to 2

**Comment by lgalis** [comment]
@martinpovolny - feedback addressed

**Pull request merged**